nexus/db-queries/src/db/datastore/crucible_dataset.rs

```diff
@@ -245,6 +245,44 @@ impl DataStore {
 
         Ok(physical_disk.disk_policy == PhysicalDiskPolicy::InService)
     }
+
+    #[cfg(any(test, feature = "testing", feature = "omdb"))]
+    pub async fn mark_crucible_dataset_not_provisionable(
+        &self,
+        opctx: &OpContext,
+        dataset_id: DatasetUuid,
+    ) -> Result<(), Error> {
+        let conn = self.pool_connection_authorized(opctx).await?;
+
+        use nexus_db_schema::schema::crucible_dataset::dsl;
+
+        diesel::update(dsl::crucible_dataset)
+            .filter(dsl::id.eq(to_db_typed_uuid(dataset_id)))
```

Comment:
Is it worth checking `time_deleted` still being NULL? Here and below

Reply:
yep! 46cc30b0e